### PR TITLE
Interactive Trust Graph

### PR DESCRIPTION
**Pull request by xoriole**
This PR includes the Trust Graph with basic interaction like node selection and dragging the graph at root node. The details of the selected node is also shown at the bottom of the graph. Here is a screenshot showing how it looks.

trust_graph_interactive

### CODE REVIEW

**Comment by MaxVanDeursen** [comment]
Dear @xoriole, I just noticed this PR and it seems to be like you're doing a continuation/rework on a project of a project of mine and a few fellow students, which can be found in #2958. Maybe you can take some inspiration from this PR, or even adjust the code to work with the current version of the Tribler codebase.

A live demo can be found [here](http://tribler.mualo.nl/).

**Comment by synctext** [comment]
Hey @MaxVanDeursen,
Yes, this is based on all the prior trust work that has been done. Key point of improvement is that this is hand-crafted QT for real-time animation. We removed the javascript dependency and installer size increase.

**Comment by MaxVanDeursen** [comment]
@synctext Sounds good! I'll keep an eye on this PR!

**Comment by xoriole** [comment]
Dear @MaxVanDeursen, I have seen your work on #2905. That is beautiful and a lot more interactive compared to this one but we wanted to have something operational with limited Qt dependencies. One of the downsides here is that the extensive CPU usage in rendering. Hopefully, it will get better with the next iteration.

File: TriblerGUI/widgets/trustgraphpage.py [hunk 1]
```diff
@@ -231,5 +342,12 @@ def update_gui_labels(self, data):
         else:
             self.window().trust_graph_progress_bar.setHidden(False)
             self.window().trust_graph_progress_bar.setValue(bootstrap_progress)
-        status_message = "Transactions: %s | Peers: %s" % (data['num_tx'], len(data['positions']))
+
+        status_message = u"<strong style='font-size:14px'>Transactions : %s &nbsp;&nbsp; " \
```
Comment (qstokkink) [inline TriblerGUI/widgets/trustgraphpage.py]: Why `&nbsp;` here and `SPACE` in the other locations?

Comment (xoriole) [reply TriblerGUI/widgets/trustgraphpage.py]: Inconsistency! Resolved

**Comment by ichorid** [comment]
Maybe you could try to use http://www.pyqtgraph.org/ to make it more efficient?

**Comment by xoriole** [comment]
@ichorid Yes, it seems like an efficient alternative. As long as we can ship it nicely in all platforms, we can use it. I'll check it out. Thanks

**Comment by xoriole** [comment]
It is working fine so I would say lets merge it for now and `pyqtgraph`, if can be used with packaging issue resolved, can come in a separate PR.

**Pull request merged**